src/beast/beast/hash/hash_append.h

```diff
@@ -21,125 +21,165 @@
 {
 };
 
-#if ! BEAST_NO_IS_CONTIGUOUS_HASHABLE_PAIR
-static_assert (is_contiguously_hashable <std::pair <
-    unsigned long long, long long>>::value, "");
-#endif
+// is_uniquely_represented<std::tuple<T...>>
 
-#if ! BEAST_NO_TUPLE_VARIADICS
-// std::tuple
 template <class ...T>
-struct is_contiguously_hashable <std::tuple<T...>>
-    : public std::integral_constant <bool,
-        static_and <is_contiguously_hashable <T>::value...>::value && 
-        static_sum <sizeof(T)...>::value == sizeof(std::tuple<T...>)>
+struct is_uniquely_represented<std::tuple<T...>>
+    : public std::integral_constant<bool,
+            static_and<is_uniquely_represented<T>::value...>::value && 
+            static_sum<sizeof(T)...>::value == sizeof(std::tuple<T...>)>
 {
 };
-#endif
 
-// std::array
+// is_uniquely_represented<T[N]>
+
 template <class T, std::size_t N>
-struct is_contiguously_hashable <std::array<T, N>>
-    : public std::integral_constant <bool,
-        is_contiguously_hashable<T>::value && 
-        sizeof(T)*N == sizeof(std::array<T, N>)>
+struct is_uniquely_represented<T[N]>
+    : public std::integral_constant<bool, is_uniquely_represented<T>::value>
 {
 };
 
-static_assert (is_contiguously_hashable <std::array<char, 3>>::value, "");
+// is_uniquely_represented<std::array<T, N>>
 
-#if ! BEAST_NO_IS_CONTIGUOUS_HASHABLE_TUPLE
-static_assert (is_contiguously_hashable <
-    std::tuple <char, char, short>>::value, "");
-#endif
+template <class T, std::size_t N>
+struct is_uniquely_represented<std::array<T, N>>
+    : public std::integral_constant<bool, is_uniquely_represented<T>::value && 
+                                          sizeof(T)*N == sizeof(std::array<T, N>)>
+{
+};
+
+/** Metafunction returning `true` if the type can be hashed in one call.
 
+    For `is_contiguously_hashable<T>::value` to be true, then for every
+    combination of possible values of `T` held in `x` and `y`,
+    if `x == y`, then it must be true that `memcmp(&x, &y, sizeof(T))`
+    return 0; i.e. that `x` and `y` are represented by the same bit pattern.
+   
+    For example:  A two's complement `int` should be contiguously hashable.
+    Every bit pattern produces a unique value that does not compare equal to
+    any other bit pattern's value.  A IEEE floating point should not be
+    contiguously hashable because -0. and 0. have different bit patterns,
+    though they compare equal.
+*/
+/** @{ */
+template <class T, class HashAlgorithm>
+struct is_contiguously_hashable
+    : public std::integral_constant<bool, is_uniquely_represented<T>::value &&
+                                      (sizeof(T) == 1 ||
+                                       HashAlgorithm::endian == endian::native)>
+{};
+
+template <class T, std::size_t N, class HashAlgorithm>
+struct is_contiguously_hashable<T[N], HashAlgorithm>
+    : public std::integral_constant<bool, is_uniquely_represented<T>::value &&
```

Comment:
Also because we went through all the trouble to write a `T[N]` version. :) https://github.com/ripple/rippled/pull/1074/files#diff-cc5411e2b135449221560c514ea21c50R142